app/src/main/java/com/itsaky/androidide/app/LimitlessIDEActivity.kt

```diff
@@ -0,0 +1,102 @@
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *   along with AndroidIDE.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package com.itsaky.androidide.app
+
+import android.content.res.Configuration
+import android.graphics.Color
+import android.graphics.Rect
+import android.os.Build
+import android.os.Bundle
+import android.os.PersistableBundle
+import android.util.TypedValue
+import android.view.WindowManager.LayoutParams.LAYOUT_IN_DISPLAY_CUTOUT_MODE_SHORT_EDGES
+import androidx.core.view.WindowCompat
+import androidx.core.view.doOnAttach
+import com.itsaky.androidide.utils.NavigationBar
+import com.itsaky.androidide.utils.getInsets
+
+/**
+ * Same as IDEActivity but DecorFitsSystemWindows is set to false
+ * Useful for creating immersive edge-to-edge experiences
+ *
+ * @author Smooth E
+ */
+abstract class LimitlessIDEActivity(
+  private val highlightNavigationBar: Boolean = false
+) : IDEActivity() {
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    makeLimitless()
+    installOnDecorViewAttachedListener()
+  }
+
+  override fun onCreate(savedInstanceState: Bundle?, persistentState: PersistableBundle?) {
+    super.onCreate(savedInstanceState, persistentState)
+    makeLimitless()
+    installOnDecorViewAttachedListener()
+  }
+
+  private fun makeLimitless() {
+    WindowCompat.setDecorFitsSystemWindows(window, false)
+
+    paintNavigationBar()
+
+    // This removes a black strip on the side where the camera cutout is in landscape mode
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
+      window.attributes.layoutInDisplayCutoutMode = LAYOUT_IN_DISPLAY_CUTOUT_MODE_SHORT_EDGES
+    }
+  }
+
+  private fun installOnDecorViewAttachedListener() {
+    window.decorView.doOnAttach { decorView -> onInsetsUpdated(getInsets(decorView)) }
+  }
+
+  /** Called whenever insets are updated */
+  open fun onInsetsUpdated(insets: Rect) { }
+
+  private fun paintNavigationBar() {
+    window.statusBarColor = Color.TRANSPARENT
+    window.navigationBarColor = Color.TRANSPARENT
+
+    if (!highlightNavigationBar) {
+      return
+    }
+
+    val refrainHighlighting =
+      NavigationBar.getInteractionMode(this) == NavigationBar.MODE_GESTURES ||
+      resources.configuration.orientation == Configuration.ORIENTATION_LANDSCAPE
+
+    if (refrainHighlighting) {
+      return
+    }
+
+    var id = com.google.android.material.R.attr.colorSurface
+    var typedValue = TypedValue()
+    theme.resolveAttribute(id, typedValue, true)
+    window.navigationBarColor = typedValue.data
+
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
+      id = com.google.android.material.R.attr.colorOutlineVariant
+      typedValue = TypedValue()
+      theme.resolveAttribute(id, typedValue, true)
+      window.navigationBarDividerColor = typedValue.data
+    }
+  }
```

Comment:
To resolve the attribute resources, you can simply use the utility method defined here : https://github.com/AndroidIDEOfficial/AndroidIDE/blob/a9a24242af34ed0cab06d7479c42b05ee410b821/common/src/main/java/com/itsaky/androidide/utils/ResourceUtils.kt#L34-L43